octodns/provider/fastdns.py

```diff
@@ -0,0 +1,533 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from requests import Session
+from akamai.edgegrid import EdgeGridAuth
+from urlparse import urljoin
+from collections import defaultdict
+
+from logging import getLogger
+from ..record import Record
+from .base import BaseProvider
+
+
+class AkamaiClientException(Exception):
+
+    _errorMessages = {
+        404: "404: Resource not found"
+    }
+
+    def __init__(self, resp):
+        try:
+            message = self._errorMessages[resp.status_code]
+
+        except KeyError:
+            resp.raise_for_status()
+
+        super(AkamaiClientException, self).__init__(message)
+
+
+class AkamaiClient(object):
+    '''
+    Client for making calls to Akamai Fast DNS API using Python Requests
+
+    Fast DNS Zone Management API V2, found here:
+    developer.akamai.com/api/web_performance/fast_dns_zone_management/v2.html
+
+    Info on Python Requests library:
+    https://2.python-requests.org/en/master/
+
+    '''
+
+    def __init__(self, _client_secret, _host, _access_token, _client_token):
+
+        self.base = "https://" + _host + "/config-dns/v2/"
+
+        sess = Session()
+        sess.auth = EdgeGridAuth(
+            client_token=_client_token,
+            client_secret=_client_secret,
+            access_token=_access_token
+        )
+        self._sess = sess
+
+    def _request(self, method, path, params=None, data=None, v1=False):
+
+        url = urljoin(self.base, path)
+        resp = self._sess.request(method, url, params=params, json=data)
+
+        if resp.status_code > 299:
+            raise AkamaiClientException(resp)
```

Comment:
I believe the way this works now (Having AkamaiClientException's __init__ do the raise_for_status) will result in slightly confusing stack traces. It's probably better to do something like:

```python
if resp.status_code == 404:
    raise AkamaiClientException()
resp.raise_for_status()
```

So that the exception is thrown in the case where it makes sense and the normal requests process happens otherwise. It might make sense to rename `AkamaiClientException` to `AkamaiNotFound` so something like that so it's clear what that exception means.